### PR TITLE
Add a TypeSystemSwiftTypeRef::GetTupleElementName() method.

### PR DESCRIPTION
**Pull request by adrian-prantl**

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test

File: lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp [hunk 1]
```diff
@@ -203,6 +203,30 @@ ResolveTypeAlias(SwiftASTContext *module_holder,
...
+  if (child->getNumChildren() != 1 &&
+      child->getKind() != Node::Kind::TupleElement)
+    return "";
+  for (NodePointer name : *child) {
+    if (name->getKind() != Node::Kind::TupleElementName)
+      continue;
+    return name->getText().str();
+  }
+  std::string name;
+  llvm::raw_string_ostream(name) << idx;
```
Comment (kastiglione) [inline lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:226]: What's the use case for returning the index if it has no name?

File: lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.h [hunk 2]
```diff
@@ -258,6 +258,10 @@ class TypeSystemSwiftTypeRef : public TypeSystemSwift {
       bool print_help_if_available, bool print_extensions_if_available,
       lldb::DescriptionLevel level = lldb::eDescriptionLevelFull) override;
 
+  /// Return the nth tuple element's name, if it has one.
```
Comment (kastiglione) [inline lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.h:261]:
```suggestion
  /// Return the nth tuple element's name, or its index as a string if it has no name.
```

**Review by kastiglione — APPROVED** [review]
Where will this be used?

File: lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp (hunk 1 above)
Comment (adrian-prantl) [reply lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:226]: This function is needed by TypeSystemSwiftTypeRef::GetChildCompilerTypeAtIndex().
The implicit name of an unnamed tuple element is 0:, 1:, ... . See https://github.com/apple/llvm-project/blob/46170cc2f5a9af3db7652ef119ecee0dbc78392d/lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp#L7018

**Pull request merged**